src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultImmutableEnumerableConverter.cs

```diff
@@ -102,19 +102,16 @@ public static void RegisterImmutableCollection(Type immutableCollectionType, Typ
             options.TryAddCreateRangeDelegate(delegateKey, createRangeDelegate);
         }
 
-        public static bool IsImmutableEnumerable(Type type, out bool IsImmutableArray)
```

Comment:
There are places in the code where we were performing null checks on collections. Since value-types can't be null and the only "known" value type collection was `ImmutableArray`, it was special cased. The changes in this PR use the more appropriate and general `JsonPropertyInfo.CanBeNull` property which makes sure "unknown" value collection types are accounted for as well.